mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -17,27 +17,32 @@ pub struct PartQueue<E: std::error::Error> {
     current_part: AsyncMutex<Option<Part>>,
     receiver: Receiver<Result<Part, PrefetchReadError<E>>>,
     failed: AtomicBool,
-    downloaded: Arc<AtomicUsize>,
+    /// The total number of bytes sent to the underlying queue of `self.receiver`
+    bytes_arrived_total: Arc<AtomicUsize>,
 }
 
 /// Producer side of the queue of [Part]s.
 #[derive(Debug)]
 pub struct PartQueueProducer<E: std::error::Error> {
     sender: Sender<Result<Part, PrefetchReadError<E>>>,
-    downloaded: Arc<AtomicUsize>,
+    /// The total number of bytes sent to `self.sender`
+    bytes_sent: Arc<AtomicUsize>,
 }
 
 /// Creates an unbounded [PartQueue] and its related [PartQueueProducer].
 pub fn unbounded_part_queue<E: std::error::Error>() -> (PartQueue<E>, PartQueueProducer<E>) {
     let (sender, receiver) = unbounded();
-    let downloaded = Arc::new(AtomicUsize::new(0));
+    let bytes_counter = Arc::new(AtomicUsize::new(0));
     let part_queue = PartQueue {
         current_part: AsyncMutex::new(None),
         receiver,
         failed: AtomicBool::new(false),
-        downloaded: Arc::clone(&downloaded),
+        bytes_arrived_total: Arc::clone(&bytes_counter),
```

Comment:
nit: maybe `bytes_received`?

Reply:
well, I put some thinking in that tbh! `bytes_received` may be misleading because of `PartQueue.receiver`, which may imply that those bytes **were received** (i.e read through receiver), but what it actually tracks the number of bytes arrived to the queue (which may also not have the clearest meaning, naming is complicated)

Reply:
Well, they are the bytes received by the receiver, so it still sounds good to be. I'd rather clarify in the rustdoc than add a different name.

Reply:
I liked `bytes_downloaded`, but also not worth bikeshedding over the name.

Reply:
I'll replace with `bytes_received`, may be not completely accurate, but symmetrical to the sending part and there is a comment with a clarification